Generated TS client fails strict ESLint presets (no-this-alias, unused disables, exhaustive-deps cleanup)

## Summary

The TypeScript client emitted from `templates/_client-common.ts.tmpl` trips four rules under the common strict ESLint preset combo (`@typescript-eslint/recommended` + `eslint-plugin-react-hooks/recommended`). Because consumers typically run codegen *before* lint in their release pipelines, every build regenerates the same offending output and breaks CI.

Observed in a downstream project running `mage patchRelease` (which runs aprot codegen → `npx eslint .`):

```
frontend/src/api/client.ts
   912:11  error    Unexpected aliasing of 'this' to local variable                                                                                                                                                                                                                @typescript-eslint/no-this-alias
  1525:3   warning  Unused eslint-disable directive (no problems were reported from 'react-hooks/exhaustive-deps')
  1881:15  warning  The ref value 'tickRef.current' will likely have changed by the time this effect cleanup function runs. If this ref points to a node rendered by React, copy 'tickRef.current' to a variable inside the effect, and use that variable in the cleanup function  react-hooks/exhaustive-deps
```

Line numbers refer to the generated file; template source locations are cited below.

## Template source

All four findings originate in `templates/_client-common.ts.tmpl`.

### 1. `@typescript-eslint/no-this-alias` — template line 809

```ts
requestStream<T>(method: string, params: unknown[], options?: RequestOptions): AsyncIterable<T> {
    const client = this;
    return {
        [Symbol.asyncIterator](): AsyncIterator<T> {
            ...
        }
    };
}
```

The `const client = this` exists so the inner `[Symbol.asyncIterator]` method can reach the outer `ApiClient`. Suggested fix: use an arrow-returning factory so `this` is captured lexically, e.g.

```ts
requestStream<T>(method: string, params: unknown[], options?: RequestOptions): AsyncIterable<T> {
    // arrow captures `this` without aliasing
    return {
        [Symbol.asyncIterator]: (): AsyncIterator<T> => {
            // use `this` directly
            ...
        }
    };
}
```

Or, if the arrow rewrite is invasive, add a scoped `// eslint-disable-next-line @typescript-eslint/no-this-alias` directly in the template above the alias.

### 2. Duplicate `exhaustive-deps` disable — template lines 1422 & 1426

```ts
// eslint-disable-next-line react-hooks/exhaustive-deps
const cacheStore = useMemo(() => {
    if (!shouldCache) return null;
    return subscribeCached<TRes>(client, cacheKey, subscribeMethod!, subscribeOpts!.params);
// eslint-disable-next-line react-hooks/exhaustive-deps
}, [client, cacheKey, shouldCache]);
```

Only the comment immediately preceding the deps array actually suppresses anything; the outer one (line 1422) is dead and is flagged as `Unused eslint-disable directive`. Suggested fix: remove line 1422.

### 3. `tickRef.current` in effect cleanup — template lines 1730–1736

```ts
useEffect(() => {
    start();
    return () => {
        tickRef.current++;
        abortRef.current?.abort();
        abortRef.current = null;
    };
}, [start]);
```

`exhaustive-deps` warns that `tickRef.current` may have changed by the time the cleanup runs. The rule's literal advice (copy into a local inside the effect body) doesn't quite fit here because the intent *is* to mutate the live ref to invalidate an in-flight async loop. Suggested fix: add a scoped `// eslint-disable-next-line react-hooks/exhaustive-deps` (with a short comment explaining the intentional live-ref mutation) above the `return () => {` line in the template.

## Why this matters for consumers

- `src/api/client.ts` is auto-generated and marked "DO NOT EDIT" — consumers can't fix it locally without re-diverging on every codegen run.
- Release pipelines that run codegen before lint (the recommended order) fail hard on the error, even after successful codegen.
- The workaround today is to `globalIgnores(['src/api/**'])` in `eslint.config.js`, but this suppresses *all* lint on the client, which is more than most teams want.

## Repro

1. `npm create vite@latest my-app -- --template react-ts`
2. Add `@typescript-eslint`, `typescript-eslint`, `eslint-plugin-react-hooks` with the `recommended` presets (matches what `npm create vite` scaffolds for TS+React).
3. Generate a client via aprot codegen.
4. `npx eslint src/api/client.ts` → reproduces all four findings.

Environment: aprot current `master`, tested in a downstream Go+React project using the React client template.

## Suggested fix

Patch `templates/_client-common.ts.tmpl` for all three issues (three small edits). Happy to open a PR if useful.

--- a/example/react/client/src/api/client.ts
+++ b/example/react/client/src/api/client.ts
@@ -802,9 +802,12 @@ export class ApiClient {
      * rejected with an error and are NOT auto-resumed on reconnect.
      */
     requestStream<T>(method: string, params: unknown[], options?: RequestOptions): AsyncIterable<T> {
-        const client = this;
+        // Arrow factory captures `this` lexically — avoids aliasing `this` to a
+        // local variable (which trips @typescript-eslint/no-this-alias under
+        // strict presets). All inner closures are also arrows, so they inherit
+        // the same `this` (the ApiClient instance).
         return {
-            [Symbol.asyncIterator](): AsyncIterator<T> {
+            [Symbol.asyncIterator]: (): AsyncIterator<T> => {
                 let id = '';
                 let started = false;
                 const buffer: T[] = [];
@@ -840,9 +843,9 @@ export class ApiClient {
 
                 const cancelAtServer = () => {
                     if (id && !ended) {
-                        client.streams.delete(id);
-                        if (client.transport.isConnected()) {
-                            client.transport.send({ type: 'cancel', id });
+                        this.streams.delete(id);
+                        if (this.transport.isConnected()) {
+                            this.transport.send({ type: 'cancel', id });
                         }
                     }
                 };
@@ -850,13 +853,13 @@ export class ApiClient {
                 const ensureStarted = () => {
                     if (started) return;
                     started = true;
-                    if (!client.transport.isConnected()) {
+                    if (!this.transport.isConnected()) {
                         end(new Error('Not connected'));
                         return;
                     }
-                    id = String(++client.requestId);
-                    client.streams.set(id, { push, end });
-                    client.notifyLoadingChange();
+                    id = String(++this.requestId);
+                    this.streams.set(id, { push, end });
+                    this.notifyLoadingChange();
                     if (options?.signal) {
                         if (options.signal.aborted) {
                             cancelAtServer();
@@ -868,7 +871,7 @@ export class ApiClient {
                             end(new Error('Request aborted'));
                         }, { once: true });
                     }
-                    client.transport.send({ type: 'request', id, method, params });
+                    this.transport.send({ type: 'request', id, method, params });
                 };
 
                 return {
@@ -1392,10 +1395,12 @@ export function useQuery<TArgs extends unknown[], TRes>(
     // These hooks are always called to satisfy Rules of Hooks.
     // When caching is off they receive no-op values and are inert.
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
     const cacheStore = useMemo(() => {
         if (!shouldCache) return null;
         return subscribeCached<TRes>(client, cacheKey, subscribeMethod!, subscribeOpts!.params);
+    // subscribeMethod / subscribeOpts.params are already encoded into cacheKey,
+    // so listing them again would force the memo to re-run on identity changes
+    // even when the resolved key is unchanged.
     // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [client, cacheKey, shouldCache]);
 
@@ -1703,6 +1708,11 @@ export function useStream<TRes>(
     useEffect(() => {
         start();
         return () => {
+            // Intentionally mutates the LIVE ref instead of a captured local —
+            // that's how cleanup invalidates the in-flight async loop on
+            // unmount/restart. Copying `tickRef.current` to a local first (the
+            // rule's literal advice) would defeat the invalidation.
+            // eslint-disable-next-line react-hooks/exhaustive-deps
             tickRef.current++;
             abortRef.current?.abort();
             abortRef.current = null;

--- a/example/vanilla/client/static/api/client.ts
+++ b/example/vanilla/client/static/api/client.ts
@@ -800,9 +800,12 @@ export class ApiClient {
      * rejected with an error and are NOT auto-resumed on reconnect.
      */
     requestStream<T>(method: string, params: unknown[], options?: RequestOptions): AsyncIterable<T> {
-        const client = this;
+        // Arrow factory captures `this` lexically — avoids aliasing `this` to a
+        // local variable (which trips @typescript-eslint/no-this-alias under
+        // strict presets). All inner closures are also arrows, so they inherit
+        // the same `this` (the ApiClient instance).
         return {
-            [Symbol.asyncIterator](): AsyncIterator<T> {
+            [Symbol.asyncIterator]: (): AsyncIterator<T> => {
                 let id = '';
                 let started = false;
                 const buffer: T[] = [];
@@ -838,9 +841,9 @@ export class ApiClient {
 
                 const cancelAtServer = () => {
                     if (id && !ended) {
-                        client.streams.delete(id);
-                        if (client.transport.isConnected()) {
-                            client.transport.send({ type: 'cancel', id });
+                        this.streams.delete(id);
+                        if (this.transport.isConnected()) {
+                            this.transport.send({ type: 'cancel', id });
                         }
                     }
                 };
@@ -848,13 +851,13 @@ export class ApiClient {
                 const ensureStarted = () => {
                     if (started) return;
                     started = true;
-                    if (!client.transport.isConnected()) {
+                    if (!this.transport.isConnected()) {
                         end(new Error('Not connected'));
                         return;
                     }
-                    id = String(++client.requestId);
-                    client.streams.set(id, { push, end });
-                    client.notifyLoadingChange();
+                    id = String(++this.requestId);
+                    this.streams.set(id, { push, end });
+                    this.notifyLoadingChange();
                     if (options?.signal) {
                         if (options.signal.aborted) {
                             cancelAtServer();
@@ -866,7 +869,7 @@ export class ApiClient {
                             end(new Error('Request aborted'));
                         }, { once: true });
                     }
-                    client.transport.send({ type: 'request', id, method, params });
+                    this.transport.send({ type: 'request', id, method, params });
                 };
 
                 return {

--- a/templates/_client-common.ts.tmpl
+++ b/templates/_client-common.ts.tmpl
@@ -806,9 +806,12 @@ export class ApiClient {
      * rejected with an error and are NOT auto-resumed on reconnect.
      */
     requestStream<T>(method: string, params: unknown[], options?: RequestOptions): AsyncIterable<T> {
-        const client = this;
+        // Arrow factory captures `this` lexically — avoids aliasing `this` to a
+        // local variable (which trips @typescript-eslint/no-this-alias under
+        // strict presets). All inner closures are also arrows, so they inherit
+        // the same `this` (the ApiClient instance).
         return {
-            [Symbol.asyncIterator](): AsyncIterator<T> {
+            [Symbol.asyncIterator]: (): AsyncIterator<T> => {
                 let id = '';
                 let started = false;
                 const buffer: T[] = [];
@@ -844,9 +847,9 @@ export class ApiClient {
 
                 const cancelAtServer = () => {
                     if (id && !ended) {
-                        client.streams.delete(id);
-                        if (client.transport.isConnected()) {
-                            client.transport.send({ type: 'cancel', id });
+                        this.streams.delete(id);
+                        if (this.transport.isConnected()) {
+                            this.transport.send({ type: 'cancel', id });
                         }
                     }
                 };
@@ -854,13 +857,13 @@ export class ApiClient {
                 const ensureStarted = () => {
                     if (started) return;
                     started = true;
-                    if (!client.transport.isConnected()) {
+                    if (!this.transport.isConnected()) {
                         end(new Error('Not connected'));
                         return;
                     }
-                    id = String(++client.requestId);
-                    client.streams.set(id, { push, end });
-                    client.notifyLoadingChange();
+                    id = String(++this.requestId);
+                    this.streams.set(id, { push, end });
+                    this.notifyLoadingChange();
                     if (options?.signal) {
                         if (options.signal.aborted) {
                             cancelAtServer();
@@ -872,7 +875,7 @@ export class ApiClient {
                             end(new Error('Request aborted'));
                         }, { once: true });
                     }
-                    client.transport.send({ type: 'request', id, method, params });
+                    this.transport.send({ type: 'request', id, method, params });
                 };
 
                 return {
@@ -1419,10 +1422,12 @@ export function useQuery<TArgs extends unknown[], TRes>(
     // These hooks are always called to satisfy Rules of Hooks.
     // When caching is off they receive no-op values and are inert.
 
-    // eslint-disable-next-line react-hooks/exhaustive-deps
     const cacheStore = useMemo(() => {
         if (!shouldCache) return null;
         return subscribeCached<TRes>(client, cacheKey, subscribeMethod!, subscribeOpts!.params);
+    // subscribeMethod / subscribeOpts.params are already encoded into cacheKey,
+    // so listing them again would force the memo to re-run on identity changes
+    // even when the resolved key is unchanged.
     // eslint-disable-next-line react-hooks/exhaustive-deps
     }, [client, cacheKey, shouldCache]);
 
@@ -1730,6 +1735,11 @@ export function useStream<TRes>(
     useEffect(() => {
         start();
         return () => {
+            // Intentionally mutates the LIVE ref instead of a captured local —
+            // that's how cleanup invalidates the in-flight async loop on
+            // unmount/restart. Copying `tickRef.current` to a local first (the
+            // rule's literal advice) would defeat the invalidation.
+            // eslint-disable-next-line react-hooks/exhaustive-deps
             tickRef.current++;
             abortRef.current?.abort();
             abortRef.current = null;

